packages/components/src/components/button/button.e2e-test.ts

```diff
@@ -1,66 +1,554 @@
+import { expect } from '@playwright/test';
+import { ComponentsPage, test } from '../../../config/playwright/setup';
+import StickerSheet from '../../../config/playwright/setup/utils/Stickersheet';
+import { BUTTON_COLORS, BUTTON_VARIANTS, DEFAULTS, ICON_BUTTON_SIZES, PILL_BUTTON_SIZES } from './button.constants';
 
-test.beforeEach(async ({ componentsPage }) => {
-  await componentsPage.mount({
-    html: `
-        <mdc-button />
-      `,
+type SetupOptions = {
+  componentsPage: ComponentsPage;
+  prefixIcon?: string;
+  postfixIcon?: string;
+  active?: boolean;
+  disabled?: boolean;
+  softDisabled?: boolean;
+  variant?: string;
+  size?: number;
+  color?: string;
+  children?: any;
+  ariaLabel?: string;
+  secondButtonForFocus?: boolean;
+};
+
+const setup = async (args: SetupOptions) => {
+  const { componentsPage, ...restArgs } = args;
+  if (restArgs.children) {
+    await componentsPage.mount({
+      html: `
+      ${restArgs.secondButtonForFocus ? '<div id="wrapper">' : ''}
+        <mdc-button
+          ${restArgs.active ? 'active' : ''}
+          ${restArgs.disabled ? 'disabled' : ''}
+          ${restArgs.softDisabled ? 'soft-disabled' : ''}
+          ${restArgs.variant ? `variant="${restArgs.variant}"` : ''}
+          ${restArgs.size ? `size="${restArgs.size}"` : ''}
+          ${restArgs.color ? `color="${restArgs.color}"` : ''}
+          ${restArgs.prefixIcon ? `prefix-icon="${restArgs.prefixIcon}"` : ''}
+          ${restArgs.postfixIcon ? `postfix-icon="${restArgs.postfixIcon}"` : ''}
+        >
+          ${restArgs.children}
+        </mdc-button>
+      ${restArgs.secondButtonForFocus ? '<mdc-button>Second Button</mdc-button></div>' : ''}
+        `,
+      clearDocument: true,
+    });
+  } else {
+    await componentsPage.mount({
+      html: `
+      ${restArgs.secondButtonForFocus ? '<div id="wrapper">' : ''}
+        <mdc-button
+          ${restArgs.active ? 'active' : ''}
+          ${restArgs.disabled ? 'disabled' : ''}
+          ${restArgs.softDisabled ? 'soft-disabled' : ''}
+          ${restArgs.variant ? `variant="${restArgs.variant}"` : ''}
+          ${restArgs.size ? `size="${restArgs.size}"` : ''}
+          ${restArgs.color ? `color="${restArgs.color}"` : ''}
+          ${restArgs.prefixIcon ? `prefix-icon="${restArgs.prefixIcon}"` : ''}
+          ${restArgs.postfixIcon ? `postfix-icon="${restArgs.postfixIcon}"` : ''}
+          ${restArgs.ariaLabel ? `aria-label="${restArgs.ariaLabel}"` : ''}
+        ></mdc-button>
+      ${restArgs.secondButtonForFocus ? '<mdc-button>Second Button</mdc-button></div>' : ''}
+        `,
+      clearDocument: true,
+    });
+  }
+
+  const element = restArgs.secondButtonForFocus
+    ? componentsPage.page.locator('div#wrapper')
+    : componentsPage.page.locator('mdc-button');
+  await element.waitFor();
+
+  // always return the first button:
+  const allButtons = await componentsPage.page.locator('mdc-button').all();
+  return allButtons[0];
```

Comment:
When we written the first button, we can use `first` instead of `all`.
```suggestion
  const firstButton = await componentsPage.page.locator('mdc-button').first();
  return firstButton;
```